[Bug] 요청 사항 - 회원 삭제 시, 프로젝트 멤버로 참가하고 있으면 삭제가 불가하는 버그 발견

### 📄 버그 설명

    @DeleteMapping("/admin/users/{memberId}") 를 실행시켰을 때, 유저가 프로젝트 멤버에 포함돼있으면 삭제가 안됨!!

Caused by: org.hibernate.exception.ConstraintViolationException: could not execute statement [ERROR: update or delete on table "user" violates foreign key constraint "fkd4fbsljbaf80kxino27yfbppd" on table "projectmember"
  Detail: Key (user_id)=(26) is still referenced from table "projectmember".] [delete from "user" where user_id=?]
	at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:97)
	at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:58)
	at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:108)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:197)
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.performNonBatchedMutation(AbstractMutationExecutor.java:134)
	at org.hibernate.engine.jdbc.mutation.internal.MutationExecutorSingleNonBatched.performNonBatchedOperations(MutationExecutorSingleNonBatched.java:55)
	at org.hibernate.engine.jdbc.mutation.internal.AbstractMutationExecutor.execute(AbstractMutationExecutor.java:55)
	at org.hibernate.persister.entity.mutation.AbstractDeleteCoordinator.doStaticDelete(AbstractDeleteCoordinator.java:280)
	at org.hibernate.persister.entity.mutation.AbstractDeleteCoordinator.delete(AbstractDeleteCoordinator.java:87)
	at org.hibernate.action.internal.EntityDeleteAction.execute(EntityDeleteAction.java:131)
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:644)
	at org.hibernate.engine.spi.ActionQueue.executeActions(ActionQueue.java:511)
	at org.hibernate.event.internal.AbstractFlushingEventListener.performExecutions(AbstractFlushingEventListener.java:414)
	at org.hibernate.event.internal.DefaultFlushEventListener.onFlush(DefaultFlushEventListener.java:41)
	at org.hibernate.event.service.internal.EventListenerGroupImpl.fireEventOnEachListener(EventListenerGroupImpl.java:127)
	at org.hibernate.internal.SessionImpl.doFlush(SessionImpl.java:1429)
	at org.hibernate.internal.SessionImpl.managedFlush(SessionImpl.java:491)
	at org.hibernate.internal.SessionImpl.flushBeforeTransactionCompletion(SessionImpl.java:2354)
	at org.hibernate.internal.SessionImpl.beforeTransactionCompletion(SessionImpl.java:1978)
	at org.hibernate.engine.jdbc.internal.JdbcCoordinatorImpl.beforeTransactionCompletion(JdbcCoordinatorImpl.java:439)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl.beforeCompletionCallback(JdbcResourceLocalTransactionCoordinatorImpl.java:169)
	at org.hibernate.resource.transaction.backend.jdbc.internal.JdbcResourceLocalTransactionCoordinatorImpl$TransactionDriverControlImpl.commit(JdbcResourceLocalTransactionCoordinatorImpl.java:267)
	at org.hibernate.engine.transaction.internal.TransactionImpl.commit(TransactionImpl.java:101)
	at org.springframework.orm.jpa.JpaTransactionManager.doCommit(JpaTransactionManager.java:563)
	... 104 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: update or delete on table "user" violates foreign key constraint "fkd4fbsljbaf80kxino27yfbppd" on table "projectmember"
  Detail: Key (user_id)=(26) is still referenced from table "projectmember".
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2736)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2421)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:525)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:435)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:196)
	at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:157)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:194)
	... 124 common frames omitted

### 🔄 재현 방법

1. user와 projectMember 와 서로 연관관계를 맺어주시고, user 에서 캐스케이티드 올로 설정하시면 될 거 같습니다.

### ✅ 예상 동작

정상적으로 삭제돼야 합니다.

### ❌ 실제 동작

삭제가 안됨

### 📸 스크린샷

<img width="1791" height="341" alt="Image" src="https://github.com/user-attachments/assets/c8f535fe-ecc8-40c1-a4b9-bccffb994887" />

### 🌍 환경 정보

_No response_

--- a/src/main/java/org/etmetmy/bn_server/domain/project/repository/ProjectMemberRepository.java
+++ b/src/main/java/org/etmetmy/bn_server/domain/project/repository/ProjectMemberRepository.java
@@ -24,7 +24,10 @@ public interface ProjectMemberRepository extends JpaRepository<ProjectMember, Lo
 
     // 프로젝트 + 유저 기준으로 매핑 삭제
     long deleteByProjectIdAndUserId(Long projectId, Long userId);
-    
+
+    // 유저 기준으로 모든 프로젝트 멤버 삭제
+    long deleteByUserId(Long userId);
+
     // 권한이 있는 모든 프로젝트 조회
     @Query("select pm.project.id from ProjectMember pm " +
             "where pm.user.id = :userId and pm.project.isDeleted != false")

--- a/src/main/java/org/etmetmy/bn_server/domain/user/service/UserServiceImpl.java
+++ b/src/main/java/org/etmetmy/bn_server/domain/user/service/UserServiceImpl.java
@@ -4,6 +4,7 @@ import lombok.RequiredArgsConstructor;
 import org.etmetmy.bn_server.domain.company.entity.Company;
 import org.etmetmy.bn_server.domain.company.repository.CompanyRepository;
 import org.etmetmy.bn_server.domain.company.service.CompanyService;
+import org.etmetmy.bn_server.domain.project.repository.ProjectMemberRepository;
 import org.etmetmy.bn_server.domain.user.dto.request.UserChangePasswordRequest;
 import org.etmetmy.bn_server.domain.user.dto.request.UserUpdateRequest;
 import org.etmetmy.bn_server.domain.user.dto.entity.UserDto;
@@ -30,6 +31,7 @@ public class UserServiceImpl implements UserService{
     private final CompanyRepository companyRepository;
     private final CompanyService companyService;
     private final PasswordEncoder passwordEncoder;
+    private final ProjectMemberRepository projectMemberRepository;
 
     // 회원 정보 수정 (Update)
     @Override
@@ -60,6 +62,11 @@ public class UserServiceImpl implements UserService{
                 .orElseThrow(UserNotFoundException::new);
 
         Long userId = user.getId();
+
+        // 먼저 해당 유저의 모든 프로젝트 멤버 삭제
+        projectMemberRepository.deleteByUserId(userId);
+
+        // 그 다음 유저 삭제
         userRepository.delete(user);
         return userId;
     }

